## pam_ldap-178-shadowlastChange.patch by Peter Marschall <peter@adpm.de>
##
## log failure of writing shadowLastChange to syslog

--- ./pam_ldap.c
+++ ./pam_ldap.c	2005-03-28 11:46:41.345626723 +0200
@@ -3625,6 +3625,8 @@
     }
   else
     {
+      int errcode;
+
       /* update shadowLastChange; may fail if not shadowAccount */
       snprintf (buf, sizeof buf, "%ld", time (NULL) / (60 * 60 * 24));
       strvals[0] = buf;
@@ -3637,8 +3639,14 @@
       mods[0] = &mod;
       mods[1] = NULL;
 
-      /* do this silently because it may fail */
-      (void) ldap_modify_s (session->ld, session->info->userdn, mods);
+      /* Only log errors because it may fail */
+      errcode = ldap_modify_s (session->ld, session->info->userdn, mods);
+      if (errcode != LDAP_SUCCESS)
+      {
+	syslog (LOG_WARNING,
+			"pam_ldap: Unable to update shadowLastChange attribute: %s",
+			ldap_err2string (rc));
+      }
 
       snprintf (errmsg, sizeof errmsg,
 		"LDAP password information changed for %s", username);
